oss_lib/config.py

```
@@ -119,25 +119,43 @@ def register_arguments(parser, service_prefix):
     parser.add_argument(
         "--debug",
         action="store_true",
-        default=False,
-        help="Set the logging level to DEBUG, overrides {0}_DEBUG "
-             "environment variable. (default: false).".format(service_prefix),
+        default=None,
+        help="Sets the logging level to DEBUG, overrides the {0}_DEBUG "
+             "environment variable and configuration "
+             "(possible values: true/false, others - unset)."
+             .format(service_prefix),
+    )
+    parser.add_argument(
+        "--no-debug",
+        action="store_false",
+        dest="debug",
+        default=None,
+        help="Opposite to --debug, disables the debug mode and overrides "
+             "the environment variable and a configuration value.",
     )
     return parser
 
 
+def get_debug_value(env_value):
+    if env_value == "true":
+        return True
+    elif env_value == "false":
+        return False
+    return None
```

This looks strange, how debug could be non-True and non-False in the same time.
Why not simply decide if debug is True or False, right here?


Also, just recommendation: consider checking "0" and "1", "False" and "True" as well.
Another approach is strict check - if env_value is non-empty and neither "true" nor "false" then raise an error with message that env variable is set to strange value

---

The debug environment variable can be set and it can have correct value, such as "true" and "false", incorrect value, or can be not set at all. Also, the default value is determined later in the code.

However, I agree that a validation have to be done in this function instead of silently hide all incorrect values and return None as though they are not set. I will fix it in the next patches.